pystac_client/collection_client.py

```diff
@@ -1,4 +1,15 @@
-from typing import TYPE_CHECKING, Any, Callable, Dict, Iterator, List, Optional, cast
+from typing import (
```

Comment:
```suggestion
from __future__ import annotations

from typing import (
```

This lets us get rid of the `"Foo"` type hints later down.